buffer/src/main/java/io/netty/buffer/api/BufferAllocator.java

```diff
@@ -133,6 +135,58 @@ static BufferAllocator offHeapPooled() {
      */
     Supplier<Buffer> constBufferSupplier(byte[] bytes);
 
+    /**
+     * Allocate a {@link Buffer} with the same size and contents of the given byte array.
+     * The allocated buffer is <em>NOT</em> backed by the given byte array, and changes to the contents of either
+     * will not be reflected in the other.
+     * This may throw an {@link OutOfMemoryError} if there is not enough free memory available to allocate a
+     * {@link Buffer} of the requested size.
+     * <p>
+     * The allocated buffer will use big endian byte order.
+     *
+     * @param bytes The byte array that determines the size and contents of the new buffer.
+     * @return The newly allocated {@link Buffer}.
+     * @throws IllegalStateException if this allocator has been {@linkplain #close() closed}.
+     */
+    default Buffer copyOf(byte[] bytes) {
+        return allocate(bytes.length).writeBytes(bytes);
+    }
+
+    /**
+     * Allocate a {@link Buffer} with the same size and contents, as the contents of the given {@link ByteBuffer}.
+     * The allocated buffer is <em>NOT</em> backed by the give byte buffer, and changes to the contents of either
+     * will not be reflected in the other.
+     * The position and limit of the given byte buffer defines the region of contents that will be copied.
+     * The position and limit are not modified by this method.
```

Comment:
This is not really correct... We modify it and so it might result in "undefined" behaviour if we use the same `ByteBuffer` as input for `copyOf` from multiple threads, even tho we reset it to the original position once done. I think we should better just call `duplicate()` and be safe here. This one extra allocation should be a good trade-off if it brings us more "safety".  Especially as `BufferAllocator` is expected to be thread-safe and used by multiple threads.